### PR TITLE
khal: override urwid

### LINKED ISSUE #282910: Build failure: khal
### Steps To Reproduce

1. `nix build nixpkgs#khal`

### Build log

https://gist.github.com/lucc/ba081193674e35e46fa8ae3a9ac02e5f

### Additional context

This might be broken since https://github.com/NixOS/nixpkgs/pull/281721 but I did not bisect.

### Notify maintainers

maintainer: @gebner 

author of #281721: @fabaff 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.72, NixOS, 24.05 (Uakari), 24.05.20240119.bbe7d8f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/11zbgb8j7wnnccbbjcq0q556h28g7p4r-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by fabaff**
## Description of changes
Fixes https://github.com/NixOS/nixpkgs/issues/282910

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 283495` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>khal</li>
    <li>khal.dist</li>
  </ul>
</details>

File: pkgs/applications/misc/khal/default.nix [hunk 1]
```diff
@@ -6,27 +6,49 @@
...
-    rev = "v${version}";
+    repo = "khal";
+    rev = "refs/tags/v${version}";
     hash = "sha256-yI33pB/t+UISvSbLUzmsZqBxLF6r8R3j9iPNeosKcYw=";
   };
 
   nativeBuildInputs = [
     glibcLocales
     installShellFiles
   ] ++ (with python3.pkgs; [
```
Comment (dotlambda) [inline pkgs/applications/misc/khal/default.nix:44]:
```suggestion
] ++ (with py.pkgs; [
```

File: pkgs/applications/misc/khal/default.nix [hunk 2]
```diff
@@ -6,27 +6,49 @@
...
+          hash = "sha256-oPb2h/+gaqkZTXIiESjExMfBNnOzDvoMkXvkZ/+KVwo=";
+        };
+        doCheck = false;
+      });
+    };
+  };
+in
+with py.pkgs;
+
+buildPythonApplication rec {
```
Comment (dotlambda) [inline pkgs/applications/misc/khal/default.nix:29]:
```suggestion
in py.pkgsbuildPythonApplication rec {
```

File: pkgs/applications/misc/khal/default.nix [hunk 3]
```diff
@@ -6,27 +6,49 @@
...
+    repo = "khal";
+    rev = "refs/tags/v${version}";
     hash = "sha256-yI33pB/t+UISvSbLUzmsZqBxLF6r8R3j9iPNeosKcYw=";
   };
 
   nativeBuildInputs = [
     glibcLocales
     installShellFiles
   ] ++ (with python3.pkgs; [
+    setuptools
```
Comment (dotlambda) [inline pkgs/applications/misc/khal/default.nix:45]: redundant if you also have setuptools-scm

Comment (fabaff) [reply pkgs/applications/misc/khal/default.nix:45]: Addresses with https://github.com/NixOS/nixpkgs/pull/284733

File: pkgs/applications/misc/khal/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/applications/misc/khal/default.nix:29]: Addresses with https://github.com/NixOS/nixpkgs/pull/284733

File: pkgs/applications/misc/khal/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/applications/misc/khal/default.nix:44]: Addresses with https://github.com/NixOS/nixpkgs/pull/284733

**Pull request merged**